fractal_server/app/routes/api/v2/task_collection.py

```diff
@@ -242,6 +253,85 @@ async def collect_tasks_pip(
     return state
 
 
+@router.post("/collect/custom/", status_code=201)
+async def collect_task_custom(
+    task_collect: TaskCollectCustomV2,
+    user: User = Depends(current_active_verified_user),
+    db: DBSyncSession = Depends(get_sync_db),
+):
+
+    settings = Inject(get_settings)
+
+    if task_collect.package_root is None:
+        if task_collect.package_name is None:
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail=(
+                    "Must provide at least one of 'package_root' "
+                    "and 'package_name'."
+                ),
+            )
+        if settings.FRACTAL_RUNNER_BACKEND == "slurm_ssh":
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail="Cannot infer 'package_root' with 'slurm_ssh' backend.",
+            )
+        res = subprocess.run(  # nosec
+            shlex.split(
+                f"{task_collect.python_interpreter} "
+                f"-m pip show {task_collect.package_name}"
+            ),
+            capture_output=True,
+            encoding="utf8",
+        )
+        package_root_dir = next(
+            (
+                it.split()[1]
+                for it in res.stdout.split("\n")
+                if it.startswith("Location")
+            ),
+            None,
+        )
+        if package_root_dir is None:
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail=(
+                    f"Package '{task_collect.package_name}' not installed at "
+                    f"{task_collect.python_interpreter}."
+                ),
+            )
+        package_root = Path(f"{package_root_dir}/{task_collect.package_name}")
+    else:
+        package_root = Path(task_collect.package_root)
+
+    task_list = _prepare_tasks_metadata(
+        package_manifest=task_collect.manifest,
+        package_source=task_collect.source,
+        python_bin=Path(task_collect.python_interpreter),
+        package_root=package_root,
+        package_version=task_collect.version,
+    )
+    # Verify that source is not already in use (note: this check is only useful
+    # to provide a user-friendly error message, but `task.source` uniqueness is
+    # already guaranteed by a constraint in the table definition).
+    for task in task_list:
+        stm = select(TaskV2).where(TaskV2.source == task.source)
```

Comment:
Could we make this a single DB statement? Would it be something like `TaskV2.source _in ["a", "b", "c]`? If it's too complex, let's stick with the current `for` loop.

Reply:
(same applies to TaskV1)